### PR TITLE
gh-101100: Fix Sphinx warnings for 2.6 deprecations and removals

### PR DESCRIPTION
**Pull request by hugovk**
Fix 54 warnings:
```
Doc/library/subprocess.rst:1464: WARNING: py:mod reference target not found: popen2
Doc/whatsnew/2.0.rst:1098: WARNING: py:mod reference target not found: linuxaudiodev
Doc/whatsnew/2.0.rst:1098: WARNING: py:mod reference target not found: sunaudiodev
Doc/whatsnew/2.0.rst:1142: WARNING: py:mod reference target not found: ihooks
Doc/whatsnew/2.2.rst:146: WARNING: py:mod reference target not found: posixfile
Doc/whatsnew/2.4.rst:386: WARNING: py:mod reference target not found: popen2
Doc/whatsnew/2.4.rst:395: WARNING: py:mod reference target not found: popen2
Doc/whatsnew/2.4.rst:395: WARNING: py:class reference target not found: Popen
Doc/whatsnew/2.4.rst:1532: WARNING: py:func reference target not found: dircache.listdir
Doc/whatsnew/2.5.rst:1681: WARNING: py:mod reference target not found: dl
Doc/whatsnew/2.5.rst:1879: WARNING: py:mod reference target not found: md5
Doc/whatsnew/2.5.rst:1879: WARNING: py:mod reference target not found: sha
Doc/whatsnew/2.5.rst:1885: WARNING: py:mod reference target not found: md5
Doc/whatsnew/2.5.rst:1885: WARNING: py:mod reference target not found: sha
Doc/whatsnew/2.6.rst:2917: WARNING: py:attr reference target not found: message
Doc/whatsnew/2.6.rst:2917: WARNING: py:attr reference target not found: args
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: audiodev
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: bgenlocations
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: buildtools
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: bundlebuilder
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: Canvas
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: compiler
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: dircache
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: dl
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: fpformat
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: gensuitemodule
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: ihooks
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: imageop
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: imgfile
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: linuxaudiodev
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: mhlib
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: mimetools
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: multifile
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: new
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: pure
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: statvfs
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: sunaudiodev
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: test.testall
Doc/whatsnew/2.6.rst:2927: WARNING: py:mod reference target not found: toaiff
Doc/whatsnew/2.6.rst:2952: WARNING: py:mod reference target not found: gopherlib
Doc/whatsnew/2.6.rst:2954: WARNING: py:mod reference target not found: MimeWriter
Doc/whatsnew/2.6.rst:2954: WARNING: py:mod reference target not found: mimify
Doc/whatsnew/2.6.rst:2958: WARNING: py:mod reference target not found: md5
Doc/whatsnew/2.6.rst:2961: WARNING: py:mod reference target not found: posixfile
Doc/whatsnew/2.6.rst:2964: WARNING: py:mod reference target not found: popen2
Doc/whatsnew/2.6.rst:2967: WARNING: py:mod reference target not found: rgbimg
Doc/whatsnew/2.6.rst:2969: WARNING: py:mod reference target not found: sets
Doc/whatsnew/2.6.rst:2972: WARNING: py:mod reference target not found: sha
Doc/whatsnew/2.7.rst:1318: WARNING: py:mod reference target not found: ihooks
Doc/whatsnew/2.7.rst:1318: WARNING: py:mod reference target not found: ihooks
Doc/whatsnew/3.0.rst:558: WARNING: py:mod reference target not found: gopherlib
Doc/whatsnew/3.0.rst:558: WARNING: py:mod reference target not found: md5
Doc/whatsnew/3.0.rst:629: WARNING: py:mod reference target not found: sets
Doc/whatsnew/3.0.rst:651: WARNING: py:mod reference target not found: new
```

* Issue: gh-101100

----
📚 Documentation preview 📚: https://cpython-previews--113725.org.readthedocs.build/

### CODE REVIEW

File: Doc/whatsnew/2.0.rst [hunk 1]
```diff
@@ -1140,7 +1140,7 @@ module.
   supported by the :mod:`gzip` module) (Contributed by James C. Ahlstrom.)
 
 * :mod:`imputil`: A module that provides a simpler way for writing customized
```
Comment (AlexWaygood) [inline Doc/whatsnew/2.0.rst]:
```suggestion
* :mod:`!imputil`: A module that provides a simpler way for writing customized
```

File: Doc/whatsnew/2.2.rst [hunk 2]
```diff
@@ -143,7 +143,7 @@ To make the set of types complete, new type objects such as :func:`dict` and
            return fcntl.lockf(self.fileno(), operation,
                               length, start, whence)
 
-The now-obsolete :mod:`posixfile` module contained a class that emulated all of
+The now-obsolete :mod:`!posixfile` module contained a class that emulated all of
 a file object's methods and also added a :meth:`lock` method, but this class
```
Comment (AlexWaygood) [inline Doc/whatsnew/2.2.rst]:
```suggestion
a file object's methods and also added a :meth:`!lock` method, but this class
```

File: Doc/whatsnew/2.6.rst [hunk 3]
```diff
@@ -2916,60 +2916,60 @@ Deprecations and Removals
 
 * Changes to the :class:`Exception` interface
   as dictated by :pep:`352` continue to be made.  For 2.6,
-  the :attr:`message` attribute is being deprecated in favor of the
-  :attr:`args` attribute.
+  the :attr:`!message` attribute is being deprecated in favor of the
+  :attr:`!args` attribute.
```
Comment (AlexWaygood) [inline Doc/whatsnew/2.6.rst]: Wait, I think the `args` attribute still exists

File: Doc/whatsnew/2.7.rst [hunk 4]
```diff
@@ -1315,8 +1315,8 @@ changes, or look through the Subversion logs for all the details.
   giving the source address that will be used for the connection.
   (Contributed by Eldon Ziegler; :issue:`3972`.)
 
-* The :mod:`ihooks` module now supports relative imports.  Note that
-  :mod:`ihooks` is an older module for customizing imports,
+* The :mod:`!ihooks` module now supports relative imports.  Note that
+  :mod:`!ihooks` is an older module for customizing imports,
   superseded by the :mod:`imputil` module added in Python 2.0.
```
Comment (AlexWaygood) [inline Doc/whatsnew/2.7.rst]:
```suggestion
  superseded by the :mod:`!imputil` module added in Python 2.0.
```

File: Doc/whatsnew/2.6.rst (hunk 3 above)
Comment (AlexWaygood) [reply Doc/whatsnew/2.6.rst]:
```suggestion
  :attr:`~BaseException.args` attribute.
```

Comment (hugovk) [reply Doc/whatsnew/2.6.rst]: Ah of course, that's where it is! Thanks!

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Pull request merged**